### PR TITLE
Send scan data to the Event Portal via MOP messages.

### PR DESCRIPTION
**Pull request by AHabes**
### What is the purpose of this change?
Send the scan data to EP via MOP messages.

### How was this change implemented?
Add the Scan data message processor, publisher, and route builder.

### How was this change tested?
Manual + Unit tests.

### Is there anything the reviewers should focus on/be aware of?
N/A

### CODE REVIEW

**Review by gregmeldrum — APPROVED** [review]
Looks good!

File: service/application/src/main/java/com/solace/maas/ep/runtime/agent/processor/ScanDataProcessor.java [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+        String scanId = (String) properties.get(RouteConstants.SCAN_ID);
+        String dataCollectionType = (String) properties.get(RouteConstants.SCAN_TYPE);
+
+        ScanDataMessage scanDataMessage =
+                new ScanDataMessage(orgId, scanId, dataCollectionType, body, Instant.now().toString());
+
+        topicDetails.put("orgId", orgId);
+        topicDetails.put("runtimeAgentId", runtimeAgentId);
+        topicDetails.put("messagingServiceId", messagingServiceId);
+        topicDetails.put(" scanId", scanId);
```
Comment (gregmeldrum) [inline service/application/src/main/java/com/solace/maas/ep/runtime/agent/processor/ScanDataProcessor.java:53]: Is the space intentional? I see it on the put and get.

**Review by gregmeldrum — APPROVED** [review]
Looks good. Just one question about the encoding/decoding of the topic details.

File: service/application/src/main/java/com/solace/maas/ep/runtime/agent/processor/ScanDataProcessor.java (hunk 1 above)
Comment (AHabes) [reply service/application/src/main/java/com/solace/maas/ep/runtime/agent/processor/ScanDataProcessor.java]: Thanks for the catch, the space has been removed.

**Pull request merged**